src/Network/Ethereum/Web3/Provider.hs

```diff
@@ -35,18 +35,14 @@ import           Network.HTTP.Client.TLS    (tlsManagerSettings)
 import           Network.HTTP.Client        (defaultManagerSettings)
 #endif
 
-import           Network.JsonRpc.TinyClient (Remote, ServerUri)
-
 -- | Any communication with Ethereum node wrapped with 'Web3' monad
-newtype Web3 a = Web3 { unWeb3 :: ReaderT (ServerUri, Manager) IO a }
+newtype Web3 a = Web3 { unWeb3 :: ReaderT (Provider, Manager) IO a }
     deriving (Functor, Applicative, Monad, MonadIO, MonadThrow)
 
-instance MonadReader (ServerUri, Manager) Web3 where
+instance MonadReader (Provider, Manager) Web3 where
     ask = Web3 ask
     local f = Web3 . local f . unWeb3
 
-instance FromJSON a => Remote Web3 (Web3 a)
```

Comment:
It's instance for `Web3` type and it should be defined in file of type declaration.